Dockerfile

```diff
@@ -19,46 +46,41 @@ ENV PYTHONUNBUFFERED=1\
     PIP_NO_CACHE_DIR=off \
     # Save runtime because it is not look for updating pip version
     PIP_DISABLE_PIP_VERSION_CHECK=on \
-    PIP_DEFAULT_TIMEOUT=100
-
-USER root
+    PIP_DEFAULT_TIMEOUT=100 \
+    # Disable poetry interaction
+    POETRY_NO_INTERACTION=1 \
+    POETRY_VIRTUALENVS_IN_PROJECT=1 \
+    POETRY_VIRTUALENVS_CREATE=1 \
+    POETRY_CACHE_DIR=/tmp/poetry_cache
 
 WORKDIR /app
 
-ENV PYTHONPATH "${PYTHONPATH}:/app"
-
-RUN set -ex \
-    # Create a non-root user
-    && addgroup --system --gid 1001 appgroup \
-    && adduser --system --uid 1001 --gid 1001 --no-create-home appuser
+# Install Poetry. Fix poetry version to avoid breaking changes
+RUN pip install poetry==1.5.1
 
-    # Upgrade the package index and install security upgrades
-RUN --mount=type=cache,target=/var/lib/apt/lists \
-    --mount=type=cache,target=/var/cache,sharing=locked \
-    apt-get update \
-    && apt-get upgrade --assume-yes \
-    && apt-get install --assume-yes --no-install-recommends python3-pip
+# Install the app. Just copy the files needed to install the dependencies
+COPY pyproject.toml poetry.lock ./
 
-# Install dependencies
-FROM python-base as poetry
-
-WORKDIR /app
-# that contains poetry=<version> to install a specific version of Poetry.
-COPY requirements.txt ./
-RUN --mount=type=cache,target=/root/.cache python3.11 -m pip install --disable-pip-version-check --requirement=requirements.txt
+#Poetry cache is used to avoid installing the dependencies every time the code changes, we will keep this folder in development environment and remove it in production
```

Comment:
```suggestion
# Poetry cache is used to avoid installing the dependencies every time the code changes, we will keep this folder in development environment and remove it in production
```